cmd/du-main.go

```diff
@@ -0,0 +1,185 @@
+/*
+ * MinIO Client (C) 2019 MinIO, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cmd
+
+import (
+	"fmt"
+	"strings"
+
+	neturl "net/url"
+
+	humanize "github.com/dustin/go-humanize"
+	"github.com/fatih/color"
+	"github.com/minio/cli"
+	json "github.com/minio/mc/pkg/colorjson"
+	"github.com/minio/mc/pkg/console"
+	"github.com/minio/mc/pkg/probe"
+)
+
+// du specific flags.
+var (
+	duFlags = []cli.Flag{
+		cli.IntFlag{
+			Name:  "max-depth, d",
+			Usage: "print the total for a folder prefix only if it is N or fewer levels below the command line argument",
+		},
+		cli.BoolFlag{
+			Name:  "human-readable, H",
+			Usage: "print sizes in human readable format (e.g., 1K 234M 2G)",
+		},
+	}
+)
+
+// Summarize disk usage.
+var duCmd = cli.Command{
+	Name:   "du",
+	Usage:  "Summarize disk usage folder prefixes recursively.",
+	Action: mainDu,
+	Before: setGlobalsFromContext,
+	Flags:  append(append(duFlags, ioFlags...), globalFlags...),
+	CustomHelpTemplate: `NAME:
+  {{.HelpName}} - {{.Usage}}
+
+USAGE:
+  {{.HelpName}} [FLAGS] TARGET
+
+FLAGS:
+  {{range .VisibleFlags}}{{.}}
+  {{end}}
+ENVIRONMENT VARIABLES:
+   MC_ENCRYPT_KEY: list of comma delimited prefix=secret values
+
+EXAMPLES:
+   1. Summarize disk usage of 'jazz-songs' bucket recursively.
+      $ {{.HelpName}} s3/jazz-songs
+
+   2. Summarize disk usage of 'louis' prefix in 'jazz-songs' bucket recursively.
+      $ {{.HelpName}} -H s3/jazz-songs/louis/
+`,
+}
+
+// Structured message depending on the type of console.
+type duMessage struct {
+	Prefix string `json:"prefix"`
```

Comment:
We seemed to use `Status      string  `json:"status"`` everrywhere, so in the case of json, I think applications will look for `status` to be `success`

Reply:
I believe you are asking to add an additional field called `"status"` with value `"success"`

Reply:
It is an mc json API behavior for all mc commands, check this with other commands..

This is the common guidance for all commands given by @abperiasamy

Reply:
Got it. Why [rmMessage](https://github.com/minio/mc/blob/master/cmd/rm-main.go#L120) doesn't have `status`? Bug in `mc` somewhere or am I missing something?

Reply:
It is a bug..